### PR TITLE
Upgraded wide_deep model for Tf 2.0

### PR DESCRIPTION
**Pull request by Atom-101**
The model uses canned estimators. The tensorflow_estimators package obtainable from pip didn't work with Tf 2.0. Building the latest version of the estimators repository from source is required.

Had to remove train hook from model training code. Keeping it raises a KeyError .

Example error for deep model:
`"The name 'dnn/head/truediv:0' refers to a Tensor which does not exist. The operation, 'dnn/head/truediv', does not exist in the graph." `

The final accuracy for census model is in line with what is mentioned in the README.
The final evaluation for a census wide model:
`I0401 18:03:14.976323 139656691826816 logger.py:147] Benchmark metric: {'name': 'accuracy', 'value': 0.8340396881103516, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.976269Z', 'extras': []}`

`I0401 18:03:14.976451 139656691826816 logger.py:147] Benchmark metric: {'name': 'accuracy_baseline', 'value': 0.7637737393379211, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.976425Z', 'extras': []}`

`I0401 18:03:14.976567 139656691826816 logger.py:147] Benchmark metric: {'name': 'auc', 'value': 0.8789075613021851, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.976542Z', 'extras': []}`

`I0401 18:03:14.976674 139656691826816 logger.py:147] Benchmark metric: {'name': 'auc_precision_recall', 'value': 0.6867934465408325, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.976653Z', 'extras': []}`

`I0401 18:03:14.976778 139656691826816 logger.py:147] Benchmark metric: {'name': 'average_loss', 'value': 0.359136164188385, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.976758Z', 'extras': []}`

`I0401 18:03:14.976883 139656691826816 logger.py:147] Benchmark metric: {'name': 'label/mean', 'value': 0.23622627556324005, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.976863Z', 'extras': []}`

`I0401 18:03:14.976987 139656691826816 logger.py:147] Benchmark metric: {'name': 'loss', 'value': 14.331116676330566, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.976967Z', 'extras': []}`

`I0401 18:03:14.977092 139656691826816 logger.py:147] Benchmark metric: {'name': 'precision', 'value': 0.6924629807472229, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.977071Z', 'extras': []}`

`I0401 18:03:14.977226 139656691826816 logger.py:147] Benchmark metric: {'name': 'prediction/mean', 'value': 0.22958678007125854, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.977175Z', 'extras': []}`

`I0401 18:03:14.977340 139656691826816 logger.py:147] Benchmark metric: {'name': 'recall', 'value': 0.5351014137268066, 'unit': None, 'global_step': 32580, 'timestamp': '2019-04-01T12:33:14.977319Z', 'extras': []}`

### CODE REVIEW

**Comment by tfboyd** [comment]
Cool I will try to take a look.  No plans to do the same work and I hope to get back to review soon.  Thank you and I hope we find something cool you would like to work on.

**Comment by tfboyd** [comment]
@tanzhenyu will be the main reviewer and will handle this once canned estimators work in TF 2.0.  Target is 15-APR.

**Comment by Atom-101** [comment]
@tanzhenyu has the canned estimators been upgraded for TF 2.0? I tried building the current estimators repository from source. It doesn't work with the tensorflow 2.0 available on pip. It gives the following error:
`AttributeError: module 'tensorflow.python.util.deprecation' has no attribute 'HIDDEN_ATTRIBUTE' `

Do I need to build the main Tensorflow repository from source?

**Comment by yhliang2018** [comment]
@Atom-101 Thanks for the PR! Looks like this PR overlaps a lot with the official/wide_deep model: https://github.com/tensorflow/models/tree/master/official/wide_deep. Could you submit your commit based on the original code? Then we can see the difference. Another question, which environment did you use to run the code? tf-nightly-2.0-preview would be the best to test CannedEstimatorV2, as we are actively changing the V2 code.

**Comment by Atom-101** [comment]
@yhliang2018 I changed the original code. 
I was able to build the estimators repository with tf-nightly-2.0-preview(I was initially using tensorflow==2.0.0-alpha0). 
Train hooks are not working because the tensors being logged do not exist in Tf 2.0(the code is trying to access the tensors by name, but the naming is now different due to eager execution).

File: official/wide_deep/census_main.py [hunk 1]
```diff
@@ -111,6 +111,6 @@ def main(_):
 
 
 if __name__ == '__main__':
-  tf.logging.set_verbosity(tf.logging.INFO)
+  tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.INFO)
```
Comment (tfboyd) [inline official/wide_deep/census_main.py]: There must be a V2 way to do logging not with tf.compat.v1.  The other person reviewing should know the right call.  If not, oh well.

File: official/wide_deep/census_main.py [hunk 2]
```diff
@@ -111,6 +111,6 @@ def main(_):
...
   define_census_flags()
-  absl_app.run(main)
+absl_app.run(main)
```
Comment (tfboyd) [inline official/wide_deep/census_main.py]: I think this needs indented 2 spaces

File: official/wide_deep/census_test.py [hunk 3]
```diff
@@ -158,4 +158,4 @@ def test_end_to_end_wide_deep(self):
 
 
 if __name__ == '__main__':
-  tf.test.main()
+tf.test.main()
```
Comment (tfboyd) [inline official/wide_deep/census_test.py]: I think this needs indented 2 spaces

File: official/wide_deep/movielens_main.py [hunk 4]
```diff
@@ -110,6 +111,6 @@ def main(_):
 
 
 if __name__ == '__main__':
-  tf.logging.set_verbosity(tf.logging.INFO)
+  tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.INFO)
   define_movie_flags()
-  absl_app.run(main)
+absl_app.run(main)
```
Comment (tfboyd) [inline official/wide_deep/movielens_main.py]: I think this needs indented 2 spaces

File: official/wide_deep/movielens_test.py [hunk 5]
```diff
@@ -114,4 +114,4 @@ def test_end_to_end_deep(self):
 
 
 if __name__ == "__main__":
-  tf.test.main()
+tf.test.main()
```
Comment (tfboyd) [inline official/wide_deep/movielens_test.py]: I think this needs indented 2 spaces

**Comment by tfboyd** [comment]
If the upgrade is to V2, it is better to nest a folder in wide_deep called (v2) rather than a new top level folder.  I added a few minor comments where I think some indents were missed.  The comment on the logging is only if there is an obvious replacement.

**Comment by yhliang2018** [comment]
@Atom-101 Thanks for trying out the nightly-preview. For the tensors to be logged, I guess you can try to put these tensors into the graph context? as looks like it's meaningless to access a tensor's name in Eager mode, and Estimator works in graph mode anyway.

**Comment by yhliang2018** [comment]
@Atom-101 Hi, we plan to have a performance test on wide_deep_model v2, and it would be nice if your PR gets checked in soon. Could you look at the comments? Let us know if something is unclear when you address it.

**Comment by Atom-101** [comment]
> @Atom-101 Hi, we plan to have a performance test on wide_deep_model v2, and it would be nice if your PR gets checked in soon. Could you look at the comments? Let us know if something is unclear when you address it.

@yhliang2018 I apologise for the delay. The training hooks for the model are trying to access the average loss through a divide operation and total loss through a sum operation. Since the model is training correctly without them, I think these hooks are for logging/displaying training progress only. The training code for tf.estimator  already logs losses internally so, maybe that can be accessed. 

[https://github.com/tensorflow/estimator/blob/9374304b02bd327126ba69c29f0d76353389357a/tensorflow_estimator/python/estimator/estimator.py#L366-L369](url)

I have not understood what you had meant by putting these tensors in the graph context in your previous comment. TF 2.0 is not building a graph is it?

**Comment by jaeyounkim** [comment]
wide_deep is deprecated.

**Pull request closed without merging**